### PR TITLE
chore(fmt/linters): Add gofumpt formatter and linter

### PR DESCRIPTION
**Pull request by walldiss**
This PR enables stricter formatting by integrating `gofumpt`, which imposes more rules than the default gofmt, allowing for improved code formatting. The results, visible in this PR, include minor yet significant cleanups such as the removal of extra new lines and the consistent declaration of variables and constants.Integrating `gofumpt` will be a valuable addition to our code formatting automation, ensuring higher code quality and consistency.Additionally, the PR updates the Makefile and linter rules to include this enhanced formatting and applies it to the codebase.

### CODE REVIEW

File: api/docgen/openrpc.go [hunk 1]
```diff
@@ -92,7 +92,7 @@ func ParseCommentsFromNodebuilderModules(moduleNames ...string) (Comments, Comme
 	return nodeComments, permComments
 }
 
-func NewOpenRPCDocument(comments Comments, permissions Comments) *go_openrpc_reflect.Document {
+func NewOpenRPCDocument(comments, permissions Comments) *go_openrpc_reflect.Document {
```
Comment (Wondertan) [inline api/docgen/openrpc.go:95]: Why though? I like that you dont need to specify the  same type over in parameters. Can we disable this rule?

File: libs/keystore/fs_keystore.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ type fsKeystore struct {
 // NewFSKeystore creates a new Keystore over OS filesystem.
 // The path must point to a directory. It is created automatically if necessary.
 func NewFSKeystore(path string, ring keyring.Keyring) (Keystore, error) {
-	err := os.Mkdir(path, 0755)
+	err := os.Mkdir(path, 0o755)
```
Comment (Wondertan) [inline libs/keystore/fs_keystore.go:27]: Wat does o mean?

Comment (walldiss) [reply libs/keystore/fs_keystore.go:27]: Octal literals instead of plain int

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
Actually there is a conflict

**Review by Wondertan — APPROVED** [review]

**Review by vgonkivs — APPROVED** [review]

**Pull request merged**